src/go/rpk/pkg/cli/debug/bundle/bundle_k8s_linux.go

```diff
@@ -381,6 +384,18 @@ func saveClusterAdminAPICalls(ctx context.Context, ps *stepParams, fs afero.Fs,
 	}
 }
 
+func saveDataplaneAPICalls(ctx context.Context, ps *stepParams, fs afero.Fs, p *config.RpkProfile) step {
```

Comment:
We should make this part of the saveCluster API calls:

- For k8s, we are losing the discovered addresses from `adminAddressesFromK8S` when we call just using the profile.
- It's fine to call the admin API call directly and not change based on whether the profile is from Cloud. If we are on a cloud cluster, then we must be inside the cluster itself, and we should be able to query the list of connections through the admin API without using the public API.
  - If we call it directly, we can skip the version check; for debug bundles this is fine.